src/algorithms/models/edgecost.ts

```diff
@@ -2,6 +2,22 @@
  * Edges represent a relation between two vertices.
  */
 class EdgeCost {
+  /**
+   * Combine costs of multiple edges into one
+   * @param ecs EdgeCost[] costs to be combined
+   * @return EdgeCost Combined cost of all ecs
+   */
+  static combine(...ecs: EdgeCost[]): EdgeCost {
+    const res = new EdgeCost();
+    for (const ec of ecs) {
```

Comment:
This is a perfect place to use reduce function 😄 Just look at this beauty:
```suggestion
    res.distance = ecs.reduce((acc, edgeCost) => acc + edgeCost.distance, res.distance);
```

Reply:
Wouldn't this get a bit unreadable once we add more costs?